diff --git a/make_aveo.inc b/make_aveo.inc
index fe91c2a..9f54ba8 100644
--- a/make_aveo.inc
+++ b/make_aveo.inc
@@ -1,6 +1,6 @@
 NCPP = /opt/nec/ve/bin/nc++
 NCC = /opt/nec/ve/bin/ncc
-NFORT = /opt/nec/ve/bin/nfort
+NFORT = @AVEO_NFORT@
 NAR = /opt/nec/ve/bin/nar
 GCC = gcc
 GPP = g++
diff --git a/src/Makefile b/src/Makefile
index f9e86aa..3bdbc98 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -131,12 +131,12 @@ $(BLIB)/libveo.a: $(VHLIB_OBJS) | $$(@D)/
 $(BLIBEX)/aveorun: $(BVE)/aveorun.o $(BVELIB)/libaveoVE.a $(BVELIB)/liburpcVE_omp.a | $$(@D)/
 	$(NFORT) $(NFORTFLAGS) -v -cxxlib -fopenmp -o $@ $^ \
 		$(NLDFLAGS) $(BVELIB)/libaveoVE.a $(BVELIB)/liburpcVE_omp.a \
-		-lveio -pthread
+		-lveio -pthread -rdynamic
 
 $(BLIBEX)/aveorun-ftrace: $(BVE)/aveorun-ftrace.o $(BVELIB)/libaveoVE.a $(BVELIB)/liburpcVE_omp.a | $$(@D)/
 	$(NFORT) $(NFORTFLAGS) -ftrace -cxxlib -fopenmp -o $@ $^ \
 		$(NLDFLAGS) $(BVELIB)/libaveoVE.a $(BVELIB)/liburpcVE_omp.a \
-		-lveio -pthread
+		-lveio -pthread -rdynamic
 
 $(BLIBEX)/relink_aveorun: ../scripts/relink_aveorun.in
 	sed -e "s,@libexecdir@,$(DEST)/libexec,g" -e "s,@velibdir@,$(VEDEST)/lib,g" \
